### PR TITLE
Incremental rock. #200

### PR DESCRIPTION
**Pull request by ssepeq**
Mixed results with SCFAs in OR51E2. Predicts diacetyl correctly as non-agonist. Ionones and δ-decalactone are failing to make a complete dock.

### CODE REVIEW

**Comment by ssepeq** [comment]
🟩🟩🟩

**Comment by ssepeq** [comment]
Waiting on the server to process all of the empirical ligands for OR51E1 and 51E2. It will take a while.

From then, it will be a matter of modifying the method_ php to make predictions. Then it should be possible to get an accuracy estimate for carboxylic acids in OR51 receptors. Other ligands probably dock in a different part of the binding residues area; this may also be true of the tetrapod-like OR families.

All this calculation may ultimately require a rack mount server. They make ones now whose cooling fans don't sound like a jet engine. Tried looking into cloud computing, but anything in budget would be less powerful than the cheap used desktop I've been developing this project on.

File: primarydock.config [hunk 1]
```diff
@@ -60,6 +60,18 @@ CEN RES 2.53 3.29 3.32 3.33 3.36 3.37 3.40 3.41 4.53 4.57 4.60 45.49 45.52 5.39
...
+# and/or transformations to be applied at specific nodes, without having to limit
+# them to a single active state node.
+# The below lines, if uncommented, will immediately move TMR6 2A in the +Y direction,
+# then rotate it 15 degrees for node 1.
+# Note all rotations and transformations are *cumulative*, not absolute; if given two
+# consecutive HXR lines each with a 5 degree rotation, the final rotation will be 10
+# degrees relative to the original protein structure.
+# HXR 0 TMR6 6.26 6.60 6.48 0 2 0 -0.55 0 0.84 0
+# HXR 1 TMR6 6.26 6.60 6.48 0 0 0 -0.55 0 0.84 15
+
```
Comment (primaryodors) [inline primarydock.config:74]: May want to add something about axes based on a region, bw residue, or atom.

File: src/classes/constants.h [hunk 2]
```diff
@@ -167,7 +167,7 @@
 #define drift_decay_rate 0.08
 
 // Allows full 360 degree whole molecule rotations to search for lower energy configurations.
-#define allow_mol_fullrot_iter 0
+#define allow_mol_fullrot_iter 1
```
Comment (primaryodors) [inline src/classes/constants.h:170]: Why is this enabled if axial tumble (line 157) is not?

File: src/classes/protein.cpp [hunk 3]
```diff
@@ -800,40 +800,54 @@ void Protein::add_remark(std::string new_remark)
     // TODO: Sort remarks by number, becarefuling to preserve the sequence of same numbered remarks.
 }
 
-void Protein::set_clashables()
+void Protein::set_clashables(int resno, bool recursed)
 {
```
Comment (primaryodors) [inline src/classes/protein.cpp:804]: This looks like the same change as made in #213 - perhaps want to close the other PR?

File: test/51e2_propionate.config [hunk 4]
```diff
@@ -17,13 +17,23 @@ LIG sdf/propionate.sdf
...
+PATH 3 REL 0 0 0
+PATH 4 REL 0 0 0
+PATH 5 REL 0 0 0
 
 SIZE 8.0 5.0 8.0
 
-ACVNODE 0
-ACVHXR TMR6 6.26 6.60 6.48 0 0 0 0.5015 0 0.8651 10?
+HXR 0 TMR6 6.26 6.60 6.48 0 2 0 bw 4.57 0
+HXR 1 TMR6 6.26 6.60 6.48 0 0 0 0.5015 0 0.8651 3
```
Comment (primaryodors) [inline test/51e2_propionate.config:29]: These lines should all match, either all BW or all Cartesian axis.

**Review by primaryodors — CHANGES_REQUESTED** [review]

File: src/classes/constants.h (hunk 2 above)
Comment (ssepeq) [reply src/classes/constants.h:170]: Will switch them in the morning and retest big three.

Comment (ssepeq) [reply src/classes/constants.h:170]: The TAAR test result is not as good if axial tumble is enabled and not full rotations.

**Comment by ssepeq** [comment]
🟩🟩🟩

**Review by primaryodors — APPROVED** [review]

**Pull request merged**